### PR TITLE
Modify Exporter docs

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: docs/trace/building-your-own-exporter/MyExporter.cs [hunk 1]
```diff
@@ -27,6 +27,10 @@ internal class MyExporter : ActivityExporter
     public override Task<ExportResult> ExportAsync(
         IEnumerable<Activity> batch, CancellationToken cancellationToken)
     {
+        // Exporter code which can generate further
+        // telemetry should do so inside SuppressInstrumentation
+        // scope. This suppresses telemetry from
+        // exporter's own code to avoid live-loop situation.
```
Comment (cijothomas) [inline docs/trace/building-your-own-exporter/MyExporter.cs:33]: Should we explain more here?

Comment (reyang) [reply docs/trace/building-your-own-exporter/MyExporter.cs:33]: I think we chatted about this [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1006#discussion_r465483708).

I would vote for not having comments to keep the code short (e.g. having 50 lines of code for a dummy exporter seems too much). No strong opinion though 😄

File: docs/trace/building-your-own-exporter/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Building your own Exporter
+
+## Background
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:3]: I suggest we move these to the bottom and rename the section to "References".

File: docs/trace/building-your-own-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* Exporters should inherit from `ActivityExporter` and implement `ExportAsync`
+  and `ShutdownAsync` methods. `ActivityExporter` is part of the [OpenTelemetry
+  Package](https://www.nuget.org/packages/opentelemetry).
+* Depending on user's choice and load on the application, `ExportAsync` may get
+  called with zero or more activities.
+* Exporters should expect to receive only sampled-in and ended activities.
+* Exporters must not throw.
+* Exporters should not modify activities they receive (the same activity may be
+  exported again by different exporter).
+* Exporters are expected to handle failures and destination appropriate retry
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:26]: This sentence is not clear to me.

File: docs/trace/building-your-own-exporter/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+## Building own Exporter
+
+* To export telemetry to a specific destination, custom exporters must be
+  written.
+* Exporters should inherit from `ActivityExporter` and implement `ExportAsync`
+  and `ShutdownAsync` methods. `ActivityExporter` is part of the [OpenTelemetry
+  Package](https://www.nuget.org/packages/opentelemetry).
+* Depending on user's choice and load on the application, `ExportAsync` may get
+  called with zero or more activities.
+* Exporters should expect to receive only sampled-in and ended activities.
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:22]: should expect to receive only => will only receive?

File: docs/trace/building-your-own-exporter/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+* Exporters must not throw.
+* Exporters should not modify activities they receive (the same activity may be
+  exported again by different exporter).
+* Exporters are expected to handle failures and destination appropriate retry
+  logic.
+
+## Example
+
+A sample exporter, which simply writes activity name
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:31]: The line wrapping seems to be weird.

File: docs/trace/building-your-own-exporter/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+to the console is shown [here](./MyExporter.cs).
+
+Apart from the exporter itself, you should also provide extensions methods to
+simplify adding the exporter to the `TracerProvider` as shown
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:35]: Is it TracerProvider or TracerProviderBuilder?
Or we can probably make it simple by saying "you should provide extension method as demonstrated here (actual link)".

File: docs/trace/building-your-own-exporter/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+* [ActivityExporter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#span-exporter)
+  export activities to specific destination.
+* OpenTelemetry .NET repo provides official exporters for
+  [Console](../../../src/OpenTelemetry.Exporter.Console/README.md),
+  [Jaeger](../../../src/OpenTelemetry.Exporter.Jaeger/README.md),
+  [OpenTelemetryProtocol](../../../src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md)
+  and [Zipkin](../../../src/OpenTelemetry.Exporter.Zipkin/README.md).
+
+## Building own Exporter
```
Comment (reyang) [inline docs/trace/building-your-own-exporter/README.md:13]: Remove this title as it is duplicated?

File: docs/trace/building-your-own-exporter/MyExporter.cs (hunk 1 above)
Comment (cijothomas) [reply docs/trace/building-your-own-exporter/MyExporter.cs:33]: I strongly feel we should leave some comments about the SupressInstrumentation part, as this is not a spec defined name.

File: docs/trace/building-your-own-exporter/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+* Exporters should inherit from `ActivityExporter` and implement `ExportAsync`
+  and `ShutdownAsync` methods. `ActivityExporter` is part of the [OpenTelemetry
+  Package](https://www.nuget.org/packages/opentelemetry).
+* Depending on user's choice and load on the application, `ExportAsync` may get
+  called with zero or more activities.
+* Exporters will only receive sampled-in and ended activities.
+* Exporters must not throw.
+* Exporters should not modify activities they receive (the same activity may be
+  exported again by different exporter).
+* Any retry logic that is required by the exporter is the responsibility of the
```
Comment (cijothomas) [inline docs/trace/building-your-own-exporter/README.md:14]: @reyang Does this look correct now? 

from spec:
_Any retry logic that is required by the exporter is the responsibility of the exporter. The default SDK SHOULD NOT implement retry logic, as the required logic is likely to depend heavily on the specific protocol and backend the spans are being sent to._
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#span-exporter

**Pull request merged**